src/Microsoft.Sbom.Adapters/Adapters/ComponentDetection/DotNetComponentExtensions.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Sbom.Adapters.ComponentDetection;
+
+using System;
+using Microsoft.ComponentDetection.Contracts.TypedComponent;
+using Microsoft.Sbom.Contracts;
+
+internal static class DotNetComponentExtensions
+{
+    private const string NotApplicablePropertyValue = "-";
+    private const string ComponentUnknownPropertyValue = "unknown";
+
+    public static SbomPackage ToSbomPackage(this DotNetComponent dotnetComponent, ExtendedScannedComponent component)
+    {
+        var sdkVersion = SanitizeString(dotnetComponent.SdkVersion);
```

Comment:
@grvillic, we discussed this in our standup, and there were concerns about enabling experimental code without some sort of guard. Since we don't have an existing feature flag mechanism, let's just go with an environment variable for now.

Reply:
Sounds good. I can add an environment variable to block this, and either customers enable it manually for SBOMs or until we have fully shipped SBOM team can remove the variable and onboard it.